### PR TITLE
Fix split issue when adding new files

### LINKED ISSUE #13: Split with new files
Hello and thanks for your amazing work (@rectorphp user here)! :wink:

From what I understand, the split only works with existing files on both sides. If I add a new file, the split does nothing (`[NOTE] No files to change` instead of `[NOTE] Adding git commit`). Is it possible to _force_ the indexation of new changes? 

I assume it's related to [this line](https://github.com/symplify/monorepo-split-github-action/blob/main/entrypoint.php#L82), but I wanted your point of view before submitting any changes.

### PR DESCRIPTION
**Pull request by ajardin**
Hi @TomasVotruba,

Here is the PR that should fix #13. Please let me know if you need further details. 😉

### CODE REVIEW

File: entrypoint.php [hunk 1]
```diff
@@ -79,13 +79,10 @@
...
 
-exec('git diff-index --quiet HEAD', $outputLines, $hasChangedFiles);
+exec('git status --porcelain', $changedFiles);
 
 
-// 1 = changed files
-// 0 = no changed files
-
-if ($hasChangedFiles === 1) {
+if ($changedFiles) {
```
Comment (TomasVotruba) [inline entrypoint.php:85]: Could you comment on the value here? It's very helpful to know the type or what is in it. 
In past this was very confusing to read after 12 months away from the code :)

File: entrypoint.php [hunk 2]
```diff
@@ -79,13 +79,10 @@
 // avoids doing the git commit failing if there are no changes to be commit, see https://stackoverflow.com/a/8123841/1348344
 exec_with_output_print('git status');
 
-exec('git diff-index --quiet HEAD', $outputLines, $hasChangedFiles);
+exec('git status --porcelain', $changedFiles);
```
Comment (TomasVotruba) [inline entrypoint.php:82]: Could you elaborate a comment of 1-2 sentences on why this is better than `diff-index`? I imagone someone will try to revert it, as `git diff` is more obvious choice for common cases.

Just so this line is protected with knowlege :) thank you

**Comment by ajardin** [comment]
You are absolutely right! Git can be a bit tricky sometimes... 🙂 

I've added some comments on the behaviour of these two commands, is that enough from your point of view?

**Comment by TomasVotruba** [comment]
Hi @ajardin , thanks for the ping. I've missed the notification in the big pile I get every day.

Looks good, let's give it a try :)

**Comment by TomasVotruba** [comment]
I know it's not related to your changed, but could you also make PHPStan pass? We need to respect green build on `main`

**Comment by ajardin** [comment]
> I know it's not related to your changed, but could you also make PHPStan pass? We need to respect green build on `main`

@TomasVotruba I fixed the PHPStan configuration, it should be okay now. 🙂

**Comment by TomasVotruba** [comment]
Thanks! Now it's time to test it before we release a new version

**Comment by yann-eugone** [comment]
Hi everyone ! :)

Just tested this action this weekend (on `2.0` tag) and I must say that this particular change is a must have so this action can be actually used.

Do you have an idea of when it's gonna be released ?
(no pressure guys, just asking so I can schedule more tests with this in order to change release workflow on my packages)

**Comment by TomasVotruba** [comment]
Hey, I thought this was already release. My bad :)

There it is:
https://github.com/symplify/monorepo-split-github-action/releases/tag/2.1

**Comment by yann-eugone** [comment]
It happens 🙂
Anyway, thank you for that quick move 🚀

**Pull request merged**